dotnet-engine/Yggdrasil.Engine/Types.cs

```diff
@@ -45,25 +47,35 @@ public override bool Equals(object? obj)
         var payload = (Payload)obj;
         return Value == payload.Value && PayloadType == payload.PayloadType;
     }
+
+    public override int GetHashCode()
```

Comment:
We have equals. We need hashcode as well to prevent the compiler complaining. I've picked an unchecked here, which allows the integer to wrap rather than creating an overflow exception